tests/scripts/test_shell_scripts.py

```
@@ -22,6 +22,50 @@ def test_detect_changes_help() -> None:
     assert_that(result.stdout).contains("Usage:")
 
 
+def test_resolve_pipeline_relevance_help() -> None:
+    """resolve-pipeline-relevance.sh should provide help and exit 0."""
+    script_path = Path("scripts/ci/resolve-pipeline-relevance.sh").resolve()
+    result = subprocess.run(  # nosec B603 - fixed argv run against a real binary in a controlled test; shell=False, no user shell input
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)